### PR TITLE
[Messenger] [Redis] Fix auth option wrongly considered invalid

### PR DESCRIPTION
**Pull request by chalasr**
| Q             | A
| ------------- | ---
| Branch?       | 5.2
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | -

Will also make the CI back to green on 5.2.

### CODE REVIEW

File: src/Symfony/Component/Messenger/Bridge/Redis/Tests/Transport/ConnectionTest.php [hunk 1]
```diff
@@ -191,17 +191,6 @@ public function testAuthFromOptionsAndDsn()
...
-    {
-        $redis = $this->createMock(\Redis::class);
-
-        $redis->expects($this->exactly(1))->method('auth')
-            ->with('password')
-            ->willReturn(true);
-
-        Connection::fromDsn('redis://password:localhost/queue', [], $redis);
-    }
-
```
Comment (chalasr) [inline src/Symfony/Component/Messenger/Bridge/Redis/Tests/Transport/ConnectionTest.php:204]: This test was added in #42067 but it never passed, the dsn is invalid (https://github.com/symfony/symfony/runs/3097443193#step:11:102)

File: src/Symfony/Component/Messenger/Bridge/Amqp/Tests/Transport/AmqpExtIntegrationTest.php [hunk 2]
```diff
@@ -76,20 +76,14 @@ public function testItSendsAndReceivesMessages()
         $this->assertEmpty(iterator_to_array($receiver->get()));
     }
 
-    /**
-     * @group legacy
-     * ^ for now, deprecation errors are thrown during serialization.
-     */
```
Comment (chalasr) [inline src/Symfony/Component/Messenger/Bridge/Amqp/Tests/Transport/AmqpExtIntegrationTest.php:82]: This test was broken because it uses an incomplete symfony serializer (see https://github.com/symfony/symfony/runs/3097443193#step:11:72)

**Comment by chalasr** [comment]
Green 🎉

File: src/Symfony/Component/Messenger/Bridge/Redis/Tests/Transport/ConnectionTest.php [hunk 3]
```diff
@@ -191,17 +191,6 @@ public function testAuthFromOptionsAndDsn()
...
 
-    public function testAuthAsUserInDsn()
-    {
-        $redis = $this->createMock(\Redis::class);
-
-        $redis->expects($this->exactly(1))->method('auth')
-            ->with('password')
-            ->willReturn(true);
-
-        Connection::fromDsn('redis://password:localhost/queue', [], $redis);
```
Comment (Nyholm) [inline src/Symfony/Component/Messenger/Bridge/Redis/Tests/Transport/ConnectionTest.php:202]: Please change this to: 

```
Connection::fromDsn('redis://password@localhost/queue', [], $redis);
```

File: src/Symfony/Component/Messenger/Bridge/Redis/Transport/Connection.php [hunk 4]
```diff
@@ -225,6 +225,7 @@ private static function validateOptions(array $options): void
     {
         $availableOptions = array_keys(self::DEFAULT_OPTIONS);
         $availableOptions[] = 'serializer';
+        $availableOptions[] = 'auth';
```
Comment (Nyholm) [inline src/Symfony/Component/Messenger/Bridge/Redis/Transport/Connection.php]: Why is this needed?

**Review by Nyholm — COMMENTED** [review]
Thank you. Just a minor thing and a question. =)

File: src/Symfony/Component/Messenger/Bridge/Redis/Tests/Transport/ConnectionTest.php (hunk 3 above)
Comment (chalasr) [reply src/Symfony/Component/Messenger/Bridge/Redis/Tests/Transport/ConnectionTest.php:202]: If I do, we're going to have a duplicate :) https://github.com/symfony/symfony/blob/ccc5513f547654ba0b37626d7f54fbd6a6086893/src/Symfony/Component/Messenger/Bridge/Redis/Tests/Transport/ConnectionTest.php#L161-L170

File: src/Symfony/Component/Messenger/Bridge/Redis/Transport/Connection.php (hunk 4 above)
Comment (chalasr) [reply src/Symfony/Component/Messenger/Bridge/Redis/Transport/Connection.php]: Because these options did not have a default and if they aren't added to the `$avaibleOptions` array, you get a deprecation notice (exception in 6.0). 
It turns out that we can easily add ones, I did :)

File: src/Symfony/Component/Messenger/Bridge/Redis/Tests/Transport/ConnectionTest.php (hunk 3 above)
Comment (Nyholm) [reply src/Symfony/Component/Messenger/Bridge/Redis/Tests/Transport/ConnectionTest.php:202]: Oh, Okey. I didn't check that.

File: src/Symfony/Component/Messenger/Bridge/Redis/Transport/Connection.php (hunk 4 above)
Comment (Nyholm) [reply src/Symfony/Component/Messenger/Bridge/Redis/Transport/Connection.php]: You are correct. Thank you.

**Review by Nyholm — APPROVED** [review]
Im happy when the CI is green

**Comment by chalasr** [comment]
Thanks for the review.
